src/libstore/build/worker.cc

```diff
@@ -198,8 +208,13 @@ void Worker::childTerminated(Goal * goal, bool wakeSleepers)
     if (i == children.end()) return;
 
     if (i->inBuildSlot) {
-        assert(nrLocalBuilds > 0);
-        nrLocalBuilds--;
+        if (goal->jobCategory() == JobCategory::Substitution) {
+            assert(nrSubstitutions > 0);
+            nrSubstitutions--;
+        } else {
+            assert(nrLocalBuilds > 0);
+            nrLocalBuilds--;
+        }
```

Comment:
Now that we have the enum, it is better to do a `switch` here. Note because of some `-Werror=...` you might need a `default: assert(false);`

Reply:
Even better would be something like
```
nrActiveGoals[goal->jobCategory()]--;
```